comp/core/workloadmeta/collectors/catalog-less/catalog.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package catalog is a wrapper that loads workloadmeta collectors, while having less
+// than the full set. Currently only used by the dogstatsd binary, this catalog does
+// not include the process-collector due to its increased dependency set.
+package catalog
+
+import (
+	"go.uber.org/fx"
+
+	remoteworkloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/collectors/internal/remote/workloadmeta"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+)
+
+// GetCatalog returns the set of FX options to populate the catalog
+func GetCatalog() fx.Option {
+	options := getCollectorOptions()
+
+	// remove nil options
+	opts := make([]fx.Option, 0, len(options))
+	for _, item := range options {
+		if item != nil {
+			opts = append(opts, item)
+		}
+	}
+	return fx.Options(opts...)
+}
+
+func remoteWorkloadmetaParams() fx.Option {
+	var filter *workloadmeta.Filter // Nil filter accepts everything
+
+	// Security Agent is only interested in containers
+	if flavor.GetFlavor() == flavor.SecurityAgent {
+		filter = workloadmeta.NewFilterBuilder().AddKind(workloadmeta.KindContainer).Build()
+	}
+
+	return fx.Provide(func() remoteworkloadmeta.Params {
+		return remoteworkloadmeta.Params{
+			Filter: filter,
```

Comment:
I am a bit confused as to why a filter is specified here.

From what I've seen so far about usage of workloadmeta, filters are specified by subscribing components. Here, we are not subscribing, but rather defining a catalog. So why are we specifying a filter?

Reply:
This refactoring is a fairly blind copy-and-rename of the other catalog. The `filter` here comes from the other catalog call, but now that you have pointed this out, I agree that it's probably not needed here, as this catalog should only be used by the dogstatsd binary.

Reply:
Pushed a commit that removes the filter from this catalog.